src/opm/output/eclipse/Summary.cpp

```diff
@@ -1208,25 +1208,50 @@ void eval_udq(const Opm::Schedule& schedule, std::size_t sim_step, Opm::SummaryS
     const UDQConfig& udq = schedule.getUDQConfig(sim_step);
     const auto& func_table = udq.function_table();
     UDQContext context(func_table, st);
-    std::vector<std::string> wells;
-    for (const auto& well_name : schedule.wellNames())
-        wells.push_back(well_name);
-
-    for (const auto& assign : udq.assignments(UDQVarType::WELL_VAR)) {
-        auto ws = assign.eval(wells);
-        for (const auto& well : wells) {
-            const auto& udq_value = ws[well];
-            if (udq_value)
-                st.update_well_var(well, ws.name(), udq_value.value());
+    {
+        std::vector<std::string> wells;
+        for (const auto& well_name : schedule.wellNames())
+            wells.push_back(well_name);
```

Comment:
Isn't this exactly equivalent to
```C++
std::vector<std::string> wells = schedule.wellNames();
```
?

In that case you could even make `wells` a `const` object since it's not modified later.

Reply:
Yes - I have updated.